src/ralph/dns/views.py

```diff
@@ -33,31 +44,53 @@ def get_dns_records(ipaddresses):
             ('type', 'A'),
             ('limit', 100),
             ('offset', 0)
-        ])  # TODO ?
+        ] + ipaddresses)
     )
     api_results = get_api_result(url)
     if api_results:
         for item in api_results:
             dns_records.append({
-                'pk': 1,
+                'pk': item['id'],
                 'name': item['name'],
                 'type': RecordType.from_name(item['type'].lower()).id,
                 'content': item['content'],
+                'ptr': item['auto_ptr'] == 2
```

Comment:
another thing: PTR record should be matched to A record when iterating through it, not by `auto_ptr` flag of A record